pyperf/_runner.py

```diff
@@ -496,6 +496,59 @@ def task_func(task, loops):
         task.inner_loops = inner_loops
         return self._main(task)
 
+    def bench_async_func(self, name, func, *args, **kwargs):
+        """Benchmark await func(*args)"""
+
+        inner_loops = kwargs.pop('inner_loops', None)
+        metadata = kwargs.pop('metadata', None)
+        self._no_keyword_argument(kwargs)
+
+        if not self._check_worker_task():
+            return None
+
+        if args:
+            func = functools.partial(func, *args)
+
+        def task_func(task, loops):
+            if loops != 1:
+                async def main():
+                    # use fast local variables
+                    local_timer = time.perf_counter
+                    local_func = func
+                    range_it = range(loops)
+
+                    t0 = local_timer()
+                    for _ in range_it:
+                        await local_func()
+                    dt = local_timer() - t0
+                    return dt
+            else:
+                async def main():
+                    # use fast local variables
+                    local_timer = time.perf_counter
+                    local_func = func
+
+                    t0 = local_timer()
+                    await local_func()
+                    dt = local_timer() - t0
+                    return dt
+
+            import asyncio
+            try:
```

Comment:
I would prefer to testing ``hasattr(asyncio, 'main')``, since main() can raise AttributeError.